### PR TITLE
doc: gs: align installation steps with VSC ext

### PR DESCRIPTION
**Pull request by greg-fer**
Edited installing automatically page to align it with the VSC extension. VSC-1254.

Signed-off-by: Grzegorz Ferenc <Grzegorz.Ferenc@nordicsemi.no>

### CODE REVIEW

**Review by Avalei — CHANGES_REQUESTED** [review]

**Review by kittydepa — COMMENTED** [review]
LGTM, and I agree with Deidre's comments. Will approve once those are addressed.

File: doc/nrf/gs_assistant.rst [hunk 1]
```diff
@@ -7,24 +8,28 @@ Installing automatically
...
+.. rst-class:: numbered-step
+
+Install prerequisites
+*********************
 
 Before you start setting up the toolchain, install available updates for your operating system.
 See :ref:`gs_recommended_versions` for information on the supported operating systems and Zephyr features.
 
-.. _gs_app_tcm:
+Additionally, if you are using a M1-based Mac machine, install the Universal version of SEGGER J-Link.
```
Comment (greg-fer) [inline doc/nrf/gs_assistant.rst:22]: Add "This is required for SEGGER J-Link to work correctly on Mac machines". TBC with Simen Tinderholt.

File: doc/nrf/gs_assistant.rst [hunk 2]
```diff
@@ -36,31 +41,63 @@ The app is installed on your machine, and the :guilabel:`Install` button changes
...
+Set up the preferred building method
+************************************
 
 There are two ways you can build an application:
 
-* To build with |VSC|, click on the :guilabel:`Open VS Code` button.
+* Using |VSC| and the |nRFVSC|
+* Using command line
+
+Installing |nRFVSC|
```
[the remark this answers is not in the document]
Comment (greg-fer) [reply doc/nrf/gs_assistant.rst:75]: To me yes, hence I kept the gerund form.

File: doc/nrf/gs_assistant.rst (hunk 1 above)
Comment (greg-fer) [reply doc/nrf/gs_assistant.rst]: Updated as discussed with Simen: no need to restrict this prerequisite to only Mac machines.

**Comment by greg-fer** [comment]
@kittydepa , @Avalei , please check again.

**Review by kittydepa — APPROVED** [review]

**Review by Avalei — APPROVED** [review]

File: doc/nrf/gs_assistant.rst [hunk 3]
```diff
@@ -36,31 +42,63 @@ The app is installed on your machine, and the :guilabel:`Install` button changes
...
    :alt: The Toolchain Manager window
 
    The Toolchain Manager window
 
-Click :guilabel:`SETTINGS` in the navigation bar to specify where you want to install the |NCS|.
-Then, in :guilabel:`SDK ENVIRONMENTS`, click the :guilabel:`Install` button next to the |NCS| version that you want to install.
-The |NCS| version of your choice is installed on your machine.
+#. Click :guilabel:`SETTINGS` in the navigation bar to specify where you want to install the |NCS|.
+#. In :guilabel:`SDK ENVIRONMENTS`, click the :guilabel:`Install` button next to the |NCS| version that you want to install.
+   The |NCS| version of your choice is installed on your machine.
```
Comment (peknis) [inline doc/nrf/gs_assistant.rst:61]:
```suggestion
#. In :guilabel:`SDK ENVIRONMENTS`, click the :guilabel:`Install` button next to the |NCS| version that you want to install.

   The |NCS| version of your choice is installed on your machine.
```
Just to separate the consequence from the actual task and keep the steps crisp.

File: doc/nrf/gs_assistant.rst [hunk 4]
```diff
@@ -36,31 +42,63 @@ The app is installed on your machine, and the :guilabel:`Install` button changes
...
+* Using |VSC| and the |nRFVSC|
+* Using command line
+
+Installing |nRFVSC|
+===================
+
+To build on the |nRFVSC|, complete the following steps:
+
+#. In Toolchain Manager, click the :guilabel:`Open VS Code` button.
+   A notification appears with a list of missing extensions that you need to install, including those from the `nRF Connect for Visual Studio Code`_ extension pack.
```
Comment (peknis) [inline doc/nrf/gs_assistant.rst:82]:
```suggestion
#. In Toolchain Manager, click the :guilabel:`Open VS Code` button.

   A notification appears with a list of missing extensions that you need to install, including those from the `nRF Connect for Visual Studio Code`_ extension pack.
```

File: doc/nrf/gs_assistant.rst [hunk 5]
```diff
@@ -36,31 +42,63 @@ The app is installed on your machine, and the :guilabel:`Install` button changes
...
+#. Click **Install missing extensions**.
+#. Once the extensions are installed, click **Open VS Code** button again.
+
+You can then follow the instructions in :ref:`gs_programming_vsc`.
+
+Installing command line requirements
+====================================
+
+To build on the command line, use the following steps:
```
Comment (peknis) [inline doc/nrf/gs_assistant.rst:91]:
```suggestion
To build on the command line, complete the following steps:
```

**Review by peknis — CHANGES_REQUESTED** [review]

File: doc/nrf/gs_assistant.rst [hunk 6]
```diff
@@ -36,31 +42,63 @@ The app is installed on your machine, and the :guilabel:`Install` button changes
...
 
 .. figure:: images/gs-assistant_tm.png
    :alt: The Toolchain Manager window
 
    The Toolchain Manager window
 
-Click :guilabel:`SETTINGS` in the navigation bar to specify where you want to install the |NCS|.
-Then, in :guilabel:`SDK ENVIRONMENTS`, click the :guilabel:`Install` button next to the |NCS| version that you want to install.
-The |NCS| version of your choice is installed on your machine.
+#. Click :guilabel:`SETTINGS` in the navigation bar to specify where you want to install the |NCS|.
```
Comment (umapraseeda) [inline doc/nrf/gs_assistant.rst:59]: There is some issue with the numbering.

File: doc/nrf/gs_assistant.rst [hunk 7]
```diff
@@ -36,31 +42,63 @@ The app is installed on your machine, and the :guilabel:`Install` button changes
...
-====================
+.. rst-class:: numbered-step
+
+Install the |NCS|
+*****************
+
+Complete the following steps to install the |NCS| source code:
 
-Once you have installed the Toolchain Manager, open it in nRF Connect for Desktop.
+1. Open Toolchain Manager in nRF Connect for Desktop.
```
Comment (umapraseeda) [inline doc/nrf/gs_assistant.rst:52]: Should we update the following image to Toolchain Manager v0.10.3 to match with the second screenshot?

File: doc/nrf/gs_programming.rst [hunk 8]
```diff
@@ -41,7 +41,7 @@ After completing the :ref:`manual <build_environment_cli>` or :ref:`automatic <g
 
 1.    Open a terminal window.
```
Comment (umapraseeda) [inline doc/nrf/gs_programming.rst:42]: Line 40: Shouldn't we update the link :ref:`automatic <gs_app_installing-ncs-tcm>` to point to https://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-8771/nrf/gs_assistant.html#installing-command-line-requirements?

File: doc/nrf/gs_assistant.rst (hunk 6 above)
Comment (greg-fer) [reply doc/nrf/gs_assistant.rst:59]: Done, alignment was missing.

**Review by peknis — APPROVED** [review]

File: doc/nrf/releases/release-notes-changelog.rst [hunk 9]
```diff
@@ -364,6 +364,7 @@ cJSON
 Documentation
 =============
 
+* Updated :ref:`gs_assistant` steps to reflect the fact that the |nRFVSC| is the default recommended IDE.
 * Updated :ref:`ug_matter_gs_adding_cluster` documentation with new code snippets to align it with the source code of refactored Matter template sample.
```
Comment (umapraseeda) [inline doc/nrf/releases/release-notes-changelog.rst:368]:
```suggestion
* Updated:

  * Steps in :ref:`gs_assistant` page to reflect the fact that the |nRFVSC| is the default recommended IDE.
  * ref:`ug_matter_gs_adding_cluster` documentation with new code snippets to align it with the source code of refactored Matter template sample.
```

**Review by umapraseeda — APPROVED** [review]

**Pull request merged**